packages/server/utils/OpenAIServerManager.ts

```diff
@@ -43,107 +41,101 @@ class OpenAIServerManager {
+      return null
+    }
   }
 
-  async groupReflections(reflectionsText: string[]) {
+  async groupReflections(reflectionsText: string[], themes: string[]) {
     if (!this.openAIApi) return null
-    const prompt = `You are given a list of reflections from a meeting, separated by commas. Your task is to group these reflections into topics and return an array of JavaScript objects. Each object represents a topic and contains an array of reflections that belong to that topic. Don't change the input text at all, even if it's spelt incorrectly or has special characters. Groups should have at least 2 reflections.
 
-    Example Input:
-    ['The retreat went well', 'The project might not be ready in time', 'The deadline is really tight', 'I liked that the length of the retreat', 'I enjoyed the hotel', 'Feeling overworked']
+    const getThemeForReflection = async (
+      reflection: string,
+      retry = false
+    ): Promise<string | null> => {
+      const prompt = `Given the themes ${themes.join(
+        ', '
+      )}, and the following reflection: "${reflection}", classify the reflection into the theme it fits in best. The reflection can only be added to one theme. Do not edit the reflection text. Your output should just be the theme name, and must be one of the themes I've provided.`
 
-    Example Output:
-    [
-    {
-    "The Retreat": [
-      "The retreat went well",
-      "I liked that the length of the retreat",
-      "I enjoyed the hotel"
-    ]
-    },
-    {
-    "Deadlines": [
-      "Feeling overworked",
-      "The project might not be ready in time",
-      "The deadline is really tight"
-    ]
-    }
-    ]
+      const response = await this.openAIApi!.createChatCompletion({
+        model: 'gpt-4',
+        messages: [
+          {
+            role: 'user',
+            content: prompt
+          }
+        ],
+        temperature: 0.7,
+        top_p: 1,
+        frequency_penalty: 0,
+        presence_penalty: 0
+      })
 
-    In the output, "The Retreat" and "Deadlines" are example topic names that you can create to group the reflections.
-    Do not give me code to run. Give me the output as valid JSON in the format above.
+      const theme = (response.data.choices[0]?.message?.content?.trim() as string) ?? null
+      if (!theme || !themes.includes(theme)) {
+        if (!retry) {
+          return getThemeForReflection(reflection, true)
+        } else {
+          sendToSentry(
+            new Error(
+              `Couldn't find a suitable theme for reflection: "${reflection}" from the following themes: ${themes.join(
+                ', '
+              )}`
+            )
+          )
+          return null
+        }
+      }
+      return theme
+    }
 
-    Here is the list of reflections: """
-    ${reflectionsText}
-    """
-    `
+    try {
+      const promises = reflectionsText.map((reflection) => getThemeForReflection(reflection))
+      const themesByReflections = await Promise.all(promises)
```

Comment:
+1 can we move `await Promise.all` in the line above to get rid of the `promises` variable?